### PR TITLE
added support for npm ci and yarn install

### PR DESCRIPTION
**Pull request by joaoagoncalves**

### CODE REVIEW

File: install/package.management.js [hunk 1]
```diff
@@ -333,22 +413,22 @@ module.exports = function (gulpWrapper, ctx) {
...
+	* Installation Task
+	* Remarks: only needs to be run the first time or after a git pull
+	*/
+	gulp.task('install', function (callback) {
 		var taskArray = [];
 		var link = pluginYargs.link == null || pluginYargs.link === true;
 		var shouldRemoveLinksBeforeInstall = link && ctx.type !== "webApp";
 
 		// Clean tasks
-		if (pluginYargs.clean !== false) {
```
Comment (francisco-maciel) [inline install/package.management.js]: we should not remove the behavior of the old "--clean" flag

File: install/package.management.js [hunk 2]
```diff
@@ -333,22 +413,22 @@ module.exports = function (gulpWrapper, ctx) {
...
+		if (pluginYargs.oldInstall) {
```
Comment (francisco-maciel) [inline install/package.management.js]: rename to "rebuildDependencies"

**Comment by francisco-maciel** [comment]
Missing version bump

File: install/package.management.js [hunk 3]
```diff
@@ -364,10 +445,13 @@ module.exports = function (gulpWrapper, ctx) {
 		}
 
 		// Add tasks
-		taskArray.push('__npmInstall');
-		// if (ctx.type === "webApp") {
-			taskArray.push('__dedupeLibs');
-		// }
+		if (pluginYargs.yarn) {
+			taskArray.push('__yarnInstall');
```
Comment (jpsfs) [inline install/package.management.js]: Are we assuming that Yarn is installed manually?

Comment (joaoagoncalves) [reply install/package.management.js]: yes, it was assumed that yarn was already installed, I will add a validation that checks if yarn is present and if it is not present maybe throw an exeception?

**Pull request merged**